pkgs/build-support/trivial-builders/default.nix

```
@@ -349,25 +356,100 @@ rec {
 
   */
   writeShellApplication =
-    { name
-    , text
-    , runtimeInputs ? [ ]
-    , meta ? { }
-    , checkPhase ? null
-    , excludeShellChecks ? [ ]
-    }:
-    writeTextFile {
-      inherit name meta;
+    let
+      runtimeShell' = runtimeShell;
+    in
+    {
+      /* The name of the script to write.
+
```

```suggestion
      /*
        The name of the script to write.

```

This would make comments easier to edit IMO. Combined with one [sentence per line](https://github.com/NixOS/nixpkgs/blob/master/doc/README.md#documentation-conventions) one could move whole sentences around without hassle.